src/lib/cap.c

```diff
@@ -200,6 +202,78 @@ void build_cap_str(cap_param *new, cap_param *old, const char *delim,
 	}
 }
 
+/** only retrun in string format for output.
+ *
+ * @param cap		capability mask.
+ * @param buf		merged capabilities in string format.
+ * @param len		max size of buf string.
+ */
+void cap_param2str(cap_param *cap, const char *delim,
+		char *buf, int len)
+{
+	unsigned int i;
+	unsigned int j = 0;
+	int r;
+	char *sp, *ep;
+	
+	unsigned int l = 0;
+	char lc[64] = "\0";
```

Comment:
Do you realize that you initialize the string with two \0 not just one? An empty string "" already ends with \0